### ISSUE TITLE
Rails STI subclasses being "forgotten" when env auto-reloads

### ISSUE THREAD

**Issue by fbatista** [rails/spring#340]
Consider the following structure:

```
class Section < ActiveRecord::Base
  belongs_to :sectionable, polymorphic: true
end

class MarketSection < Section
end

class MarketNotesSection < MarketSection
end

```

When doing a query for sectionables that have `has_many :market_sections, as: :sectionable` , on the first run after the environment is loaded, it correctly performs `sections.type IN ('MarketSection', 'MarketNotesSection')`, because i added an initializer to preload the classes in development, as suggested in http://www.alexreisner.com/code/single-table-inheritance-in-rails for example.

However, as soon as i edit and save a bit of code (a controller for example), the environment "forgets" about the subclass of marketsection and stops including it in the query.

I'd be happy to provide more details if needed.

**Comment by lostapathy** [comment]
This is apparently still an issue, but I think it's an issue in rails not spring.  I ran into the issue in an app that did not use spring.

**Comment by akaspick** [comment]
@lostapathy This just bit me as well.  The suggested ```to_prepare``` solution seemed to work for me.

**Comment by jrochkind** [comment]
Yeah, I think it is a general Rails issue, and is kind of sort of documented in the autoloading Guide. It is a gotcha with STI and dev-mode auto-loading. 

https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#autoloading-and-sti
